### PR TITLE
More http --> https updates

### PR DESCRIPTION
**Pull request by SeanKilleen**
## Summary
All links tested first by hand. Anomalous looking changes are noted in PR review comments. 

Supports https://github.com/dotnet/docs/issues/4504

### CODE REVIEW

**Comment by SeanKilleen** [comment]
I also created https://github.com/App-vNext/Polly/issues/718 since the Polly folks have several links in the docs.

**Comment by SeanKilleen** [comment]
Created https://github.com/IdentityServer/IdentityServer4/issues/3783 in case the IdentityServer folks want to make use of the SSL cert that readthedocs.io will provision for custom domains.

**Comment by SeanKilleen** [comment]
This is ready for review.

File: docs/core/index.md
Comment (SeanKilleen) [inline docs/core/index.md]: ℹ️ this where the link redirects to. The original site does not support https; the redirect does. Will defer to you on whether you want to keep the catch branded URL or the https-compliant one. 👍

File: docs/machine-learning/tutorials/predict-prices-with-model-builder.md
Comment (SeanKilleen) [inline docs/machine-learning/tutorials/predict-prices-with-model-builder.md]: ℹ️ This site has been redesigned; the URL auto-forwards to this updated URL, which is https compliant.

File: docs/framework/wcf/feature-details/security-protocols-version-1-0.md
Comment (Youssef1313) [inline docs/framework/wcf/feature-details/security-protocols-version-1-0.md]: I think those should remain http as they're.

File: docs/core/index.md
Comment (mairaw) [reply docs/core/index.md]: we usually try to avoid extra redirects but the previous one does look catchy 😄 

@tmds do you know if the previous site ever had any content or was it always a redirect?

File: docs/framework/wcf/feature-details/security-protocols-version-1-0.md
Comment (mairaw) [reply docs/framework/wcf/feature-details/security-protocols-version-1-0.md]: I agree with @Youssef1313 here. Also the change from http to https in the other cells was a mistake as a well.

One option here would be to lock the URL inside backticks so it's treated as code instead of URL, or we could keep as-is knowing these represent namespace values.

File: docs/machine-learning/tutorials/predict-prices-with-model-builder.md
Comment (mairaw) [reply docs/machine-learning/tutorials/predict-prices-with-model-builder.md]: I think it's a great improvement since the site auto-redirects to this new URL.

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @SeanKilleen. Nice changes! Let's decide what to do with the namespace values but let's not upgrade them to https on the WCF topic.

File: docs/core/index.md
Comment (Youssef1313) [reply docs/core/index.md]: @mairaw, Its first appearance in the Wayback Machine in Apr 2016 was a redirect to `http://developers.redhat.com/dotnet/?utm_source=event&utm_medium=vanityurl&utm_term=redhatloves.net&utm_campaign=msbuild`

Comment (mairaw) [reply docs/core/index.md]: yep, I was checking that too. I think in this case the catchy URL could stay since it does redirect to the https one

File: docs/framework/wcf/feature-details/security-protocols-version-1-0.md
Comment (SeanKilleen) [reply docs/framework/wcf/feature-details/security-protocols-version-1-0.md]: Makes sense; will change back 👍

File: docs/framework/wcf/feature-details/security-protocols-version-1-0.md
Comment (SeanKilleen) [inline docs/framework/wcf/feature-details/security-protocols-version-1-0.md]: ℹ️ These are reverted to http per request re: schemas.

**Comment by SeanKilleen** [comment]
@mairaw should be good to go now; let me know if you need any other changes 👍

File: docs/framework/wcf/feature-details/security-protocols-version-1-0.md
Comment (mairaw) [reply docs/framework/wcf/feature-details/security-protocols-version-1-0.md]: Thank you!

**Review by mairaw — APPROVED** [review]
Looks good @SeanKilleen! Thanks.

File: docs/spark/tutorials/get-started.md
Comment (mairaw) [inline docs/spark/tutorials/get-started.md]: One last request, sorry. I was just double checking the links and this one is not working for me with https

```suggestion
    * Download [Apache Maven 3.6.2](http://mirror.metrocast.net/apache/maven/maven-3/3.6.2/binaries/apache-maven-3.6.2-bin.zip).
```

Comment (SeanKilleen) [reply docs/spark/tutorials/get-started.md]: Weird -- it definitely worked for me yesterday; even have the zip file in my downloads directory from trying it 😆 

> image

Will accept the suggestion 👍

**Comment by mairaw** [comment]
Thanks again!

**Pull request merged**
